scripts/bundles/internal-platform-hydrate.ts

```diff
@@ -39,6 +39,9 @@ export async function internalHydrate(opts: BuildOptions) {
       banner: getBanner(opts, 'Stencil Hydrate Platform'),
       preferConst: true,
     },
+    treeshake: {
+      moduleSideEffects: false,
+    },
```

Comment:
the `internal-platform-hydrate` bundle depends on functions which use the output-target utils which, in turn, imports `path`. However, none of the functions which it uses actually _use_ functioned exported by `path`, so when bundling it doesn't actually import anything from `path` but if you don't set this option Rollup will leave an unqualified import like so:

```ts
import "path";
```

if you say `moduleSideEffects: false` you're basically saying 'dont assume that imported a module has a desired side effect' which allows an import that doesn't result in any symbols being pulled into the importing module to be eliminated.

Another way to say this is that without this change the treeshaking validation for `test.dist` doesn't pass for this bundle.

Reply:
Can we capture this info in a comment for future us?